baseband/mark4/base.py

```diff
@@ -274,29 +280,30 @@ def read(self, count=None, fill_value=0., squeeze=True, out=None):
         ----------
         count : int
             Number of samples to read.  If omitted or negative, the whole
-            file is read.
+            file is read.  Ignored if ``out`` is given.
         fill_value : float
             Value to use for invalid or missing data.
-        squeeze : bool
-            If `True` (default), remove channel and thread dimensions if unity.
         out : `None` or array
-            Array to store the data in. If given, count will be inferred,
-            and squeeze is set to `False`.
+            Array to store the data in. If given, ``count`` will be inferred.
+            If ``squeeze`` is `True`, ``out`` must have squeezed sample
+            dimensions.
 
         Returns
         -------
         out : array of float
-            Dimensions are (sample-time, vlbi-thread, channel).
+            Dimensions are (sample-time, channel) with dimensions
+            of length unity possibly removed (if ``squeeze`` is `True`).
         """
         if out is None:
             if count is None or count < 0:
                 count = self.size - self.offset
 
-            out = np.empty((self.nthread, count),
-                           dtype=self._frame.dtype).T
+            result = np.empty((self._sample_shape.nchan, count),
```

Comment:
Maybe just `self._sample_shape + (count,)` for consistency with dada?

Reply:
It also looks nicer and is shorter, so changed for all formats except VDIF.